commit/merkleroot/outcome.go

```diff
@@ -293,6 +293,7 @@ func checkForReportTransmission(
 	consensusObservation consensusObservation,
 ) Outcome {
 	offRampUpdated := false
+	// TODO: does this need to be updated to ensure all seqNum's are updated instead of just one?
```

Comment:
@dimkouv it looks like a single updated offramp allows us to transition states. If we were to continue with checking all all seqNums would that work with multiple reports?

Reply:
The plugin only ever operates on a single offramp, there is no multiple-offramp scenario.

Reply:
Isn't there a seqNum for each source chain? `OffRampNextSeqNums` is `map[cciptypes.ChainSelector]cciptypes.SeqNum`. If we split reports across 5 source chains don't we need to wait until all 5 reports transmit?

Reply:
> If we split reports across 5 source chains don't we need to wait until all 5 reports transmit?

Ideally, no. It would be better if the plugin can build on the chain(s) whose report(s) transmitted while waiting on other chains.

Reply:
> > If we split reports across 5 source chains don't we need to wait until all 5 reports transmit?
> 
> Ideally, no. It would be better if the plugin can build on the chain(s) whose report(s) transmitted while waiting on other chains.

That seems like an unnecessary optimization. AFAIK all transmits should take about the same amount of time, and may even be included in the same block. What your suggesting would require the observation phase to use the same logic we have in the checkForReportTransmission phase.

Reply:
Yeah, its an optimization, which is why I prefaced with "ideally" :-)